### ISSUE TITLE
Tax rounding problem

### ISSUE THREAD

**Issue by mdmoore** [woocommerce/woocommerce#19706]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [ ] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1.  Create 2 products with a price of $10.50
2.  Create a tax rate of 8.7%
3.  Pick only one of the 2 products, but add a quantity of 2 of them to the cart
4. Check tax amount
5. Reduce the quantity to 1
6. Add the 2nd product, created in step 1, to the cart
7. Check tax amount

## Expected/actual behavior

When I follow those steps, I see...

The products and subtotal are the same, yet the tax differs by 1 cent.  

I was expecting to see...

The same tax amount.

## Same product, Quantity 2

![http://cld.wthms.co/SRnV5l](http://cld.wthms.co/SRnV5l+)
Full Image: http://cld.wthms.co/SRnV5l+

## 2 products, same price

![http://cld.wthms.co/5YXwWu](http://cld.wthms.co/5YXwWu+)
Full Image: http://cld.wthms.co/5YXwWu+

## Isolating the problem

- [x] This bug happens with only WooCommerce plugin active
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>

```
### WordPress Environment ###

Home URL: https://woo.dev
Site URL: https://woo.dev
WC Version: 3.3.5
Log Directory Writable: ✔
WP Version: 5.0-alpha-42970
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 7.0.29
PHP Post Max Size: 30 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.59.0
OpenSSL/1.0.2o

SUHOSIN Installed: –
Max Upload Size: 25 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.3.5
WC Database Prefix: wp_
Total Database Size: 25.01MB
Database Data Size: 21.23MB
Database Index Size: 3.78MB
wp_woocommerce_sessions: Data: 0.74MB + Index: 0.04MB
wp_woocommerce_api_keys: Data: 0.00MB + Index: 0.01MB
wp_woocommerce_attribute_taxonomies: Data: 0.00MB + Index: 0.01MB
wp_woocommerce_downloadable_product_permissions: Data: 0.01MB + Index: 0.01MB
wp_woocommerce_order_items: Data: 0.06MB + Index: 0.04MB
wp_woocommerce_order_itemmeta: Data: 0.43MB + Index: 0.31MB
wp_woocommerce_tax_rates: Data: 0.00MB + Index: 0.01MB
wp_woocommerce_tax_rate_locations: Data: 0.00MB + Index: 0.01MB
wp_woocommerce_shipping_zones: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_shipping_zone_locations: Data: 0.00MB + Index: 0.01MB
wp_woocommerce_shipping_zone_methods: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_payment_tokens: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_payment_tokenmeta: Data: 0.00MB + Index: 0.01MB
wp_woocommerce_log: Data: 0.00MB + Index: 0.00MB
wp_ariadminer_connections: Data: 0.00MB + Index: 0.00MB
wp_commentmeta: Data: 0.03MB + Index: 0.02MB
wp_comments: Data: 0.76MB + Index: 0.35MB
wp_email_log: Data: 12.01MB + Index: 0.02MB
wp_followup_coupons: Data: 0.00MB + Index: 0.01MB
wp_followup_coupon_logs: Data: 0.00MB + Index: 0.00MB
wp_followup_customers: Data: 0.00MB + Index: 0.01MB
wp_followup_customer_carts: Data: 0.01MB + Index: 0.01MB
wp_followup_customer_notes: Data: 0.00MB + Index: 0.00MB
wp_followup_customer_orders: Data: 0.00MB + Index: 0.00MB
wp_followup_email_coupons: Data: 0.00MB + Index: 0.00MB
wp_followup_email_excludes: Data: 0.00MB + Index: 0.00MB
wp_followup_email_logs: Data: 0.07MB + Index: 0.03MB
wp_followup_email_orders: Data: 0.43MB + Index: 0.08MB
wp_followup_email_order_coupons: Data: 0.00MB + Index: 0.00MB
wp_followup_email_tracking: Data: 0.01MB + Index: 0.02MB
wp_followup_followup_history: Data: 0.18MB + Index: 0.02MB
wp_followup_order_categories: Data: 0.00MB + Index: 0.00MB
wp_followup_order_items: Data: 0.00MB + Index: 0.01MB
wp_followup_subscribers: Data: 0.00MB + Index: 0.01MB
wp_followup_subscribers_to_lists: Data: 0.00MB + Index: 0.00MB
wp_followup_subscriber_lists: Data: 0.00MB + Index: 0.01MB
wp_links: Data: 0.00MB + Index: 0.00MB
wp_options: Data: 2.42MB + Index: 0.11MB
wp_postmeta: Data: 3.38MB + Index: 2.16MB
wp_posts: Data: 0.56MB + Index: 0.23MB
wp_say_what_strings: Data: 0.00MB + Index: 0.00MB
wp_snippets: Data: 0.00MB + Index: 0.00MB
wp_termmeta: Data: 0.01MB + Index: 0.01MB
wp_terms: Data: 0.00MB + Index: 0.01MB
wp_term_relationships: Data: 0.06MB + Index: 0.13MB
wp_term_taxonomy: Data: 0.01MB + Index: 0.01MB
wp_usermeta: Data: 0.05MB + Index: 0.03MB
wp_users: Data: 0.00MB + Index: 0.01MB
wp_wcpv_commissions: Data: 0.00MB + Index: 0.00MB
wp_wcpv_per_product_shipping_rules: Data: 0.00MB + Index: 0.00MB
wp_wc_booking_relationships: Data: 0.00MB + Index: 0.00MB
wp_wc_deposits_payment_plans: Data: 0.00MB + Index: 0.00MB
wp_wc_deposits_payment_plans_schedule: Data: 0.00MB + Index: 0.00MB
wp_wc_download_log: Data: 0.00MB + Index: 0.00MB
wp_wc_points_rewards_user_points: Data: 0.00MB + Index: 0.00MB
wp_wc_points_rewards_user_points_log: Data: 0.00MB + Index: 0.01MB
wp_wc_warranty_products: Data: 0.00MB + Index: 0.00MB
wp_wc_webhooks: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_bundled_itemmeta: Data: 0.00MB + Index: 0.01MB
wp_woocommerce_bundled_items: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_per_product_shipping_rules: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_subscription_downloads: Data: 0.00MB + Index: 0.00MB

### Post Type Counts ###

attachment: 115
bookable_person: 5
bookable_resource: 2
course: 1
custom_css: 1
customize_changeset: 7
event_ticket: 2
event_ticket_email: 1
follow_up_email: 51
global_product_addon: 1
jetpack_migration: 2
lesson: 1
nav_menu_item: 13
page: 28
post: 6
product: 169
product_variation: 304
quiz: 1
revision: 44
scheduled-action: 373
shop_coupon: 20
shop_order: 517
shop_order_refund: 6
shop_subscription: 33
tribe_events: 1
tribe_organizer: 1
tribe_venue: 1
tribe_wooticket: 2
warranty_request: 1
wc_booking: 16
wc_membership_plan: 2
wc_user_membership: 2
wc_zapier_feed: 2
wishlist: 3

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (1) ###

WooCommerce: by Automattic – 3.3.5

### Settings ###

API Enabled: ✔
Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: accommodation-booking (accommodation-booking)
booking (booking)
bundle (bundle)
external (external)
grouped (grouped)
mix-and-match (mix-and-match)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #4 - /shop/
Cart: #5 - /cart/
Checkout: #6 - /checkout/
My account: #7 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 2.2.8
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –
```

</details>

**Comment by claudiulodro** [comment]
[Tax calculations are performed per line](https://github.com/woocommerce/woocommerce/wiki/How-Taxes-Work-in-WooCommerce#tax-calculations-simplified) in WC, so what is happening is you have something like this:

```
1 x productA = .913 tax which gets rounded down to .91
1x productB = .913 tax which gets rounded down to .91
Total tax: 1.82
```
vs.
```
2 x productA = 1.826 tax which gets rounded up to 1.83
Total tax: 1.83
```

It's a slight quirk of the tax calculation system, but is still calculating correctly.

**Comment by milo7** [comment]
Thanks, Mike, I understand your point of needing to compute taxes line by line because of multiple rates. In this case, rounding up the tax amount at the subtotal level seems to take care of the discrepancies.

I checked the documentation and  I'm impressed to see that woocommerce already offers this option:

"If you enable Rounding tax at subtotal level, instead of per line, the rounding is done at the Subtotal level."

I checked the box, and the issue is now solved.

https://docs.woocommerce.com/document/setting-up-taxes-in-woocommerce/
